pretend/src/lib.rs

```diff
@@ -334,6 +335,7 @@ impl<T> Response<T> {
 /// be implemented by this struct.
 ///
 /// See crate level documentation for more information
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
I'm not sure that you want to compare a client, so Eq, PartialEq and Hash might not make sense too much ...